src/markdown/general/mas_data_voor_vogelatlas.Rmd

```diff
@@ -0,0 +1,254 @@
+---
+title: "MAS data voor vogelatlas"
+author: "Ward Langeraert"
+date: "`r Sys.Date()`"
+output:
+  bookdown::html_document2:
+    code_folding: hide
+    toc: true
+    toc_float: true
+    toc_collapsed: true
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+renv::restore()
+library(knitr)
+library(here)
+opts_chunk$set(
+  echo = TRUE,
+  message = FALSE,
+  warning = FALSE)
+opts_knit$set(root.dir = here::here())
+```
+
+
+```{r}
+Sys.setenv("PROJ_NETWORK" = "ON")
+
+library(tidyverse)
+library(sf)
+```
+
+# MAS pilootproject
+## Inladen data
+
+We laden de data in van het MAS pilootproject.
+
+```{r}
+data_path <- here("data", "processed")
+mas_data_df <- read_csv(here(data_path, "MAS_data_full_2018_2023.csv"))
+```
+
+We selecteren vogels met broedcode groter dan nul.
+
+```{r}
+mas_piloot_broed <- mas_data_df %>%
+  filter(soortgrp == 2,
+         wrntype > 0) %>%
+  select(plotnaam, jaar, periode_in_jaar, naam, aantal, x_plot, y_plot)
+```
+
+
+## Berekening maximaal aantal
+
+We berekenen het maximum aantal individuen per jaar per telcirkel voor elke soort.
+
+```{r}
+mas_piloot_atlas <- mas_piloot_broed %>% 
+  group_by(plotnaam, jaar, periode_in_jaar, naam) %>%
+  mutate(totaal = sum(aantal)) %>%
+  group_by(plotnaam, jaar, naam, x_plot, y_plot) %>%
+  summarise(max_aantal = max(totaal), .groups = "drop") %>%
+  select(plotnaam, jaar, naam, max_aantal, x_plot, y_plot) %>%
+  mutate(x_coord = x_plot,
+         y_coord = y_plot)
+```
+
+Zet om naar sf object met telcirkels.
+
+```{r}
+mas_piloot_atlas_sf <- st_as_sf(mas_piloot_atlas,
+                                coords = c("x_plot", "y_plot"),
+                                crs = 31370) %>%
+  st_buffer(300)
+```
+
+
+# MBAG (zand)leemstreek
+## Inladen data
+
+We laden de data van 2023 in.
+
+```{r}
+data_path2 <- here("data", "mas")
+file_geojson_2023 <- file.path(data_path2, 
+                               "20230810_qgis_export_sovon_wfs_2023.geojson")
+mas_geojson_2023 <- read_sf(file_geojson_2023)
+```
+
+We selecteren de data van de zandleemstreek en de westelijke leemstreek.
+We selecteren vogels met broedcode groter dan nul.
+
+```{r}
+mbag_broed_amersfoord <- mas_geojson_2023 %>%
+  filter(soortgrp == 2,
+         wrntype > 0,
+         grepl("^(Zn|Lm)", plotnaam))
```

Comment:
werkt dit? Ik deed altijd `grepl("^(Zn|Lm)", .$plotnaam)`, maar misschien is dit ondertussen niet meer nodig. Was in elk geval vroeger nodig om grepl te laten werken in een pipe.

Reply:
Ja hoor dit werkt prima